tests/integration/cli_did_negative_test.go

```diff
@@ -0,0 +1,406 @@
+			PublicKeyMultibase: string(pubKeyMultibase582),
+		})
+		followingUpdatedPayload.Authentication = append(followingUpdatedPayload.Authentication, keyId2AsExtraController)
+		followingUpdatedPayload.CapabilityDelegation = []string{keyId}
+		followingUpdatedPayload.CapabilityInvocation = []string{keyId}
+
+		signInputsAugmented := append(signInputs, signInputs2...)
+
+		// Fail to update the DID Doc with missing cli arguments
+		//   a. missing payload, sign inputs and account
+		_, err = cli.UpdateDid(types.MsgUpdateDidPayload{}, []cli_types.SignInput{}, "")
+		Expect(err).ToNot(BeNil())
+
+		//   b. missing payload, sign inputs
+		_, err = cli.UpdateDid(types.MsgUpdateDidPayload{}, []cli_types.SignInput{}, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
+
+		//   c. missing payload, account
+		_, err = cli.UpdateDid(types.MsgUpdateDidPayload{}, signInputs, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
+
+		//   d. missing sign inputs, account
+		_, err = cli.UpdateDid(followingUpdatedPayload, []cli_types.SignInput{}, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
+
+		//   e. missing payload
+		_, err = cli.UpdateDid(types.MsgUpdateDidPayload{}, signInputs, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
+
+		//   f. missing sign inputs
+		_, err = cli.UpdateDid(followingUpdatedPayload, []cli_types.SignInput{}, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
+
+		//   g. missing account
+		_, err = cli.UpdateDid(followingUpdatedPayload, signInputs, "")
+		Expect(err).ToNot(BeNil())
+
+		// Fail to update the DID Doc with sign inputs mismatch
+		//   a. sign inputs total mismatch
+		_, err = cli.UpdateDid(followingUpdatedPayload, signInputsFuzzed, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
+
+		//   b. sign inputs invalid length
+		_, err = cli.UpdateDid(followingUpdatedPayload, signInputs, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
+
+		//   c. non-existing key id
+		_, err = cli.UpdateDid(followingUpdatedPayload, []cli_types.SignInput{
+			{
+				VerificationMethodId: keyId,
+				PrivKey:              privKey,
+			},
+			{
+				VerificationMethodId: "non-existing-key-id",
+				PrivKey:              privKey2,
+			},
+		}, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
+
+		//  d. non-matching private key
+		_, err = cli.UpdateDid(followingUpdatedPayload, []cli_types.SignInput{
+			{
+				VerificationMethodId: keyId2AsExtraController,
+				PrivKey:              privKey,
+			},
+			{
+				VerificationMethodId: keyId,
+				PrivKey:              privKey2,
+			},
+		}, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
+
+		//  e. invalid private key
+		_, err = cli.UpdateDid(followingUpdatedPayload, []cli_types.SignInput{
+			{
+				VerificationMethodId: keyId,
+				PrivKey:              privKeyFuzzedExtra,
+			},
+			{
+				VerificationMethodId: keyId2AsExtraController,
+				PrivKey:              privKey2,
+			},
+		}, testdata.BASE_ACCOUNT_1)
+		Expect(err).ToNot(BeNil())
```

Comment:
There's a lot of repetition here to provide the different inputs required, which makes the test hard to read. Will this be better off [as a table, defined once, and used across these tests](https://onsi.github.io/ginkgo/#table-specs)?